plugins/task/privacyconsent/src/Extension/PrivacyConsent.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  Task.PrivacyConsent
+ *
+ * @copyright   (C) 2023 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Plugin\Task\PrivacyConsent\Extension;
+
+use Joomla\CMS\Application\ApplicationHelper;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Mail\Exception\MailDisabledException;
+use Joomla\CMS\Mail\MailTemplate;
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\CMS\Router\Route;
+use Joomla\CMS\Uri\Uri;
+use Joomla\CMS\User\UserHelper;
+use Joomla\Component\Scheduler\Administrator\Event\ExecuteTaskEvent;
+use Joomla\Component\Scheduler\Administrator\Task\Status;
+use Joomla\Component\Scheduler\Administrator\Task\Task;
+use Joomla\Component\Scheduler\Administrator\Traits\TaskPluginTrait;
+use Joomla\Database\DatabaseAwareTrait;
+use Joomla\Database\ParameterType;
+use Joomla\Event\SubscriberInterface;
+use PHPMailer\PHPMailer\Exception as phpmailerException;
+
+
+// phpcs:disable PSR1.Files.SideEffects
+\defined('_JEXEC') or die;
+// phpcs:enable PSR1.Files.SideEffects
+
+/**
+ * A task plugin. Offers 2 task routines Invalidate Expired Consents and Remind Expired Consents
+ * {@see ExecuteTaskEvent}.
+ *
+ * @since __DEPLOY_VERSION__
+ */
+final class PrivacyConsent extends CMSPlugin implements SubscriberInterface
+{
+    use DatabaseAwareTrait;
+    use TaskPluginTrait;
+
+    /**
+     * @var string[]
+     * @since __DEPLOY_VERSION__
+     */
+    private const TASKS_MAP = [
+        'invalidate.expired' => [
+            'langConstPrefix' => 'PLG_TASK_PRIVACYCONSENT_INVALIDATE',
+            'method'          => 'invalidateExpiredConsents',
+            'form'            => 'invalidateForm',
+        ],
+        'remind.expired' => [
+            'langConstPrefix' => 'PLG_TASK_PRIVACYCONSENT_REMIND',
+            'method'          => 'remindExpiringConsents',
+            'form'            => 'remindForm',
+        ],
+    ];
+
+    /**
+     * @var boolean
```

Comment:
```suggestion
     * @var boolean

```
PHPCS empty line needed.